allennlp/data/dataset_readers/penn_tree_bank.py

```diff
@@ -0,0 +1,173 @@
+
+from collections import OrderedDict
+from typing import Dict, List
+import logging
+import os
+
+from overrides import overrides
+# NLTK is so performance orientated (ha ha) that they have lazy imports. Why? Who knows.
+from nltk.corpus.reader.bracket_parse import BracketParseCorpusReader # pylint: disable=no-name-in-module
+from nltk.tree import Tree
+
+from allennlp.common import Params
+from allennlp.common.file_utils import cached_path
+from allennlp.common.tqdm import Tqdm
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import TextField, SpanField, SequenceLabelField, ListField, Field
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Token
+from allennlp.data.dataset_readers.dataset_utils.span_utils import enumerate_spans
+
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("ptb_trees")
+class PennTreeBankDatasetReader(DatasetReader):
+    """
+    Reads constituency parses from the WSJ part of the Penn Tree Bank from the LDC.
+
```

Comment:
Can you give an overview of the input/output here?  You have the input, but you don't have the output at all, and it's not what you might expect from the name of the class.  I might also rename this reader to something that reflects that this is a span tagging task.  All of the names I can think of are super long, but `ptb_trees` is misleading, so we should find something better.